plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java

```diff
@@ -840,6 +840,26 @@ protected void verifyColumnNameLengthFailurePermissible(Throwable e)
                 .hasMessageContaining("Fields must contain only letters, numbers, and underscores, start with a letter or underscore, and be at most 300 characters long.");
     }
 
+    @Test
+    public void testCollate()
+    {
+        try (TestTable table = new TestTable(bigQuerySqlExecutor, "test.test_collate", "(col1 STRING)")) {
```

Comment:
Please create a table having collate:
```
(col1 STRING COLLATE 'und:ci')
```

Reply:
I Had tried a similar thing before and it gave the same error which I saw before.
For example:
```java
        try (TestTable table = new TestTable(bigQuerySqlExecutor, "test.test_collate", "(col1 STRING COLLATE 'und:ci')")) {
            onBigQuery(format("INSERT INTO %s VALUES ('a'), ('B'), ('b')", table.getName()));
            assertQuery(format("SELECT * FROM %s", table.getName()), "VALUES ('a'), ('B'), ('b')");
        }
```
Fails with:
`Caused by: com.google.api.gax.rpc.FailedPreconditionException: io.grpc.StatusRuntimeException: FAILED_PRECONDITION: request failed: column 'col1' is unsupported by the Storage API`

Reply:
As I replied in https://github.com/trinodb/trino/pull/15584#issuecomment-1370326626, throwing an exception is the expected behavior at this time. Please verify the failure.

Reply:
Oh ok. Let me check and verify if we can improve the error message and expect an exception to be thrown.

Reply:
@ebyhr quick inputs needed. Not sure about the concept of how this BigQuery plugin is failing here because of `assertQuery` method. It doesn't break when we run `COLLATE` inside `onBigQuery`.

The test fails when we try to run assert and not the collate SQL.
i.e., the test fails at:
`assertQuery(format("SELECT * FROM %s", table.getName()), "VALUES ('a'), ('B'), ('b')");`

The code breaks at this [line](https://github.com/trinodb/trino/blob/master/plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java#L147). Wherein, it effectively tries to get a ReadSession using `bigQueryReadClient.createReadSession(createReadSessionRequest)`

SQL which fails doesn't contain `COLLATE` keyword.

So:
- The Storage API failure fails to recognise the column name for some reason. How can we identify this exception of the column name because of COLLATE?
- Are there any examples of throwing TrinoException during Split Manager when Trino invokes a plugin?

Reply:
> Not sure about the concept of how this BigQuery plugin is failing here because of assertQuery method. It doesn't break when we run COLLATE inside onBigQuery.

`onBigQuery` uses query based API instead of storage read API. 

> The Storage API failure fails to recognise the column name for some reason. How can we identify this exception of the column name because of COLLATE?

I don't know. Let me look for when I have time.

> Are there any examples of throwing TrinoException during Split Manager when Trino invokes a plugin?

Sorry, I don't understand the question.